src/ol/array.js

```diff
@@ -1,8 +1,65 @@
 goog.provide('ol.array');
 
-goog.require('goog.array');
 goog.require('goog.asserts');
 
+function defaultCompare(a, b) {
```

Comment:
We have `ol.array.numberSafeCompareFunction` already, which you might want to use (and maybe rename to `ol.array.defaultCompare`, as [discussed](https://github.com/openlayers/ol3/pull/4570#discussion_r47946510)).